pkg/client/auth/oidc/device.go

```diff
@@ -145,5 +145,8 @@ func makeErrorForHTTPResponse(resp *http.Response) error {
 	if err != nil {
 		return err
 	}
-	return fmt.Errorf("%s %s returned HTTP %s; \n\n %#q", resp.Request.Method, resp.Request.URL, resp.Status, bodyBytes)
+	url := resp.Request.URL.String()
+	safeURL := strings.Replace(url, "\n", "", -1)
+	safeURL = strings.Replace(safeURL, "\r", "", -1)
+	return fmt.Errorf("%s %s returned HTTP %s; \n\n %#q", resp.Request.Method, safeURL, resp.Status, bodyBytes)
 }
```

Comment:
Suggest making a sanitizeUserInput function under https://github.com/G-Research/armada/tree/master/internal/common somewhere

Reply:
of course